waspc/data/Generator/templates/react-app/src/operations/resources.js

```diff
@@ -31,26 +31,26 @@ export function addResourcesUsedByQuery(queryCacheKey, resources) {
 export function getQueriesUsingResource(resource) {
   return Array.from(resourceToQueryCacheKeys.get(resource) || [])
 }
-
 /**
  * Invalidates all queries that are using specified resources.
  * @param {string[]} resources - Names of resources.
  */
-export function invalidateQueriesUsing(resources) {
-  const queryCacheKeysToInvalidate = new Set()
-  for (const resource of resources) {
-    getQueriesUsingResource(resource).forEach(key => queryCacheKeysToInvalidate.add(key))
-  }
-  for (const queryCacheKey of queryCacheKeysToInvalidate) {
+export async function invalidateQueriesUsing(resources) {
+  const queryClient = await queryClientInitialized
+
+  const queryCacheKeysToInvalidate = new Set(resources.flatMap(getQueriesUsingResource))
+  queryCacheKeysToInvalidate.forEach(queryCacheKey => 
     queryClient.invalidateQueries(queryCacheKey)
-  }
+  )
 }
 
-export function removeQueries() {
+export async function removeQueries() {
+  const queryClient = await queryClientInitialized
   queryClient.removeQueries()
 }
 
-export function invalidateAndRemoveQueries() {
+export async function invalidateAndRemoveQueries() {
+  const queryClient = await queryClientInitialized
```

Comment:
Had to make these three functions asynchronous. I checked there usages and I don't think it affected anything, but you can double check if you want.

Most of the call sites were either already `async` functions or weren't be impacted by becoming one.